alicloud/data_source_alicloud_db_instances_sql_log.go

```diff
@@ -0,0 +1,181 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/rds"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func dataSourceAlicloudDBInstancesSQLLog() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudDBInstancesSQLLogRead,
+
+		Schema: map[string]*schema.Schema{
+			"db_instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"start_time": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"end_time": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"sql_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"query_key_words": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"data_base": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"page_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(30, 100),
+				Default:      30,
+			},
+			"page_number": {
+				Type:     schema.TypeInt,
+				Optional: true,
```

Comment:
There is no need to use `page_size` and `page_number`. Please return all filtered records.